test/lib/server.test.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+const Assert = require('chai').assert;
+const mockery = require('mockery');
+const engine = require('catbox-memory');
+
+describe('server case', () => {
+    let hapiEngine;
+
+    before(() => {
+        mockery.enable({
+            warnOnUnregistered: false,
+            useCleanCache: true
+        });
+    });
+
+    beforeEach(() => {
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.resetCache();
+        hapiEngine = null;
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    describe('positive cases', () => {
+        let error;
+        let server;
+
+        before((done) => {
+            /* eslint-disable global-require */
+            hapiEngine = require('../../lib/server');
+            /* eslint-enable global-require */
+
+            hapiEngine({
+                httpd: {
+                    port: 12347
+                },
+                cache: { engine },
+                auth: {
+                    jwtPublicKey: '12345'
+                }
+            }, (e, s) => {
+                error = e;
+                server = s;
+                done();
```

Comment:
You could also pass in `error` to done and it'll fail if it's a truthy value. This would eliminate the need to `Assert.notOk(error)`